include/aws/s3/s3_client.h

```diff
@@ -403,6 +403,9 @@ struct aws_s3_meta_request_options {
      * be used. If not NULL, these options will override the client options. */
     const struct aws_signing_config_aws *signing_config;
 
+    /* The presigned url*/
+    const char *presigned_url;
```

Comment:
Correct me if I am wrong:
Looks like all you need is to skip the signing with presigned url. 

I was going to suggest that leave `signing_config` to be NULL, but as a meta request, we cannot change the client level setting. 

But, as a workaround, you can pass anonymous credentials into the `signing_config`, which should skip the signing for you. It is a workaround without any code change to the lib

Reply:
> Correct me if I am wrong: Looks like all you need is to skip the signing with presigned url.
> 
> I was going to suggest that leave `signing_config` to be NULL, but as a meta request, we cannot change the client level setting.
> 
> But, as a workaround, you can pass anonymous credentials into the `signing_config`, which should skip the signing for you. It is a workaround without any code change to the lib

May I ask how to pass anoymous credentials to the `signing_config`? 

Based on my investigation, the `aws_s3_meta_request_sign_request_default` is the function creates signing_config and pass it to meta_request. https://github.com/awslabs/aws-c-s3/pull/297/files#diff-436560b33218855638d58d1c14ecca658afd2c8ec65f2b525e21c6d829a7068aR755
so we have to bypass the following eventually which was what I did in the PR.
```
.sign_request = aws_s3_meta_request_sign_request_default,
```

Reply:
```
    struct aws_signing_config_aws config;
    AWS_ZERO_STRUCT(config);
    struct aws_credentials *credentials = aws_credentials_new_anonymous(allocator);
    config.credentials = credentials;
```

Reply:
That function does NOT create the `signing_config`, it fetches config from meta request or client. https://github.com/awslabs/aws-c-s3/pull/297/files#diff-436560b33218855638d58d1c14ecca658afd2c8ec65f2b525e21c6d829a7068aR701-R716

```
if (meta_request->cached_signing_config != NULL) {
        signing_config = meta_request->cached_signing_config->config;
    } else if (client->cached_signing_config != NULL) {
        signing_config = client->cached_signing_config->config;
    } else {
        AWS_LOGF_DEBUG(
            AWS_LS_S3_META_REQUEST,
            "id=%p: No signing config present. Not signing request %p.",
            (void *)meta_request,
            (void *)request);
        on_signing_complete(NULL, AWS_ERROR_SUCCESS, user_data);
        return;
}
```

Reply:
so if I understand right, your suggestion
```
 struct aws_signing_config_aws config;
    AWS_ZERO_STRUCT(config);
    struct aws_credentials *credentials = aws_credentials_new_anonymous(allocator);
    config.credentials = credentials;
```
should be put inside the `else` block before calling `on_signing_complete`. Am I right?